### PR TITLE
 Remove 'resource', 'clientId', 'authority' from MSIDTokenCacheKey

### PR DESCRIPTION
**Pull request by antrix1989**
See #54 for details.

Depends on #76.

### CODE REVIEW

File: IdentityCore/src/cache/MSIDTokenCacheKey.m [hunk 1]
```diff
@@ -282,9 +282,23 @@ - (void)encodeWithCoder:(NSCoder *)coder
...
+    // Backward compatibility with ADAL.
+    if (_service)
+    {
+        NSArray<NSString *> * items = [_service componentsSeparatedByString:@"|"];
+        if (items.count == 4) // See s_adalServiceFormat.
+        {
+            NSString *authority = [items[1] msidBase64UrlDecode];
+            [coder encodeObject:authority forKey:@"authority"];
+            
+            NSString *resource = items[2] == s_nilKey ? nil : [items[2] msidBase64UrlDecode];
```
Comment (oldalton) [inline IdentityCore/src/cache/MSIDTokenCacheKey.m]: shouldn't it be isEqualToString?

File: IdentityCore/src/cache/MSIDTokenCacheKey.m [hunk 2]
```diff
@@ -199,16 +193,29 @@ - (NSUInteger)hash
...
                           resource:(NSString *)resource
                           clientId:(NSString *)clientId
 {
     
-    return [NSString stringWithFormat:@"%@|%@|%@|%@",
+    return [NSString stringWithFormat:s_adalServiceFormat,
             s_adalLibraryString,
             authority.absoluteString.msidBase64UrlEncode,
-            [self.class getAttributeName:resource.msidBase64UrlEncode],
+            [self.class getAttributeName:resource],
```
Comment (oldalton) [inline IdentityCore/src/cache/MSIDTokenCacheKey.m]: 👍 thanks for catching this

File: IdentityCore/src/cache/MSIDTokenCacheKey.m [hunk 3]
```diff
@@ -221,7 +228,7 @@ + (NSString *)serviceWithAuthority:(NSURL *)authority
         return nil;
     }
     
-    return [NSString stringWithFormat:@"%@$%@$%@",
+    return [NSString stringWithFormat:s_msalServiceFormat,
```
Comment (oldalton) [inline IdentityCore/src/cache/MSIDTokenCacheKey.m]: this has changed in the new schema

File: IdentityCore/src/cache/MSIDTokenCacheKey.m (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/src/cache/MSIDTokenCacheKey.m]: yep, it should :(

File: IdentityCore/src/cache/key/MSIDLegacyTokenCacheKey.m [hunk 4]
```diff
@@ -154,9 +139,23 @@ - (void)encodeWithCoder:(NSCoder *)coder
...
+        NSArray<NSString *> * items = [_service componentsSeparatedByString:@"|"];
+        if (items.count == 4) // See s_adalServiceFormat.
+        {
+            NSString *authority = [items[1] msidBase64UrlDecode];
+            [coder encodeObject:authority forKey:@"authority"];
+            
+            NSString *resource = [items[2] isEqualToString:s_nilKey] ? nil : [items[2] msidBase64UrlDecode];
+            [coder encodeObject:resource forKey:@"resource"];
+            
+            NSString *clientId= [items[3] msidBase64UrlDecode];
```
Comment (oldalton) [inline IdentityCore/src/cache/key/MSIDLegacyTokenCacheKey.m]: nit: space =

**Pull request merged**